### PR TITLE
settings: make mocking helpers return cleanups

### PR DESCRIPTION
**Pull request by dt**

### CODE REVIEW

File: pkg/kv/txn_coord_sender_test.go [hunk 1]
```diff
@@ -1559,10 +1559,7 @@ func TestContextDoneNil(t *testing.T) {
 
 func TestTooManyIntents(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	defer func(s *settings.IntSetting) {
-		maxIntents = s
-	}(maxIntents)
-	maxIntents = settings.TestingIntSetting(3)
+	defer settings.TestingSetInt(&maxIntents, 3)
```
Comment (tamird) [inline pkg/kv/txn_coord_sender_test.go]: I think you're missing a pair of parens on the end of this.

**Comment by petermattis** [comment]
:lgtm:

---

Review status: 0 of 5 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[pkg/settings/registry.go, line 290 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/15361#-KifNglrdLXCOWT6RQk3:-KifNglrdLXCOWT6RQk4:bvx3qiz) ([raw file](https://github.com/cockroachdb/cockroach/blob/4129fe0eded7d8ad5bd34b20eb8c00331aaa0cf3/pkg/settings/registry.go#L290)):*
> ```Go
> func TestingByteSizeSetting(i int64) *ByteSizeSetting {
> 	s := &ByteSizeSetting{IntSetting{defaultValue: i}}
> 	s.setToDefault()
> ```

What was the `setToDefault` doing? Or why is it no longer necessary?

---

*[pkg/storage/replica_test.go, line 7713 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/15361#-KifNZft83gmXPR4qOKn:-KifNZft83gmXPR4qOKo:brdb3v5) ([raw file](https://github.com/cockroachdb/cockroach/blob/5f75a1e6cf6f9a9e87f29f6a06b990a746bcf2d4/pkg/storage/replica_test.go#L7713)):*
> ```Go
> 	defer leaktest.AfterTest(t)()
> 
> 	defer settings.TestingSetByteSize(&maxCommandSize, 1024)
> ```

You need to invoke the function returned by `TestSetByteSize`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15361#-:-KifNmh2TTCvBSrm6i6H:bnfp4nl)*

**Comment by knz** [comment]
LGTM modulo fixing the defers

---

Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

*[pkg/sql/logic_test.go, line 1416 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/15361#-KifNpI229iSWRrgk8xH:-KifNpI4ny5YGWiDUmK0:b-kf1ict) ([raw file](https://github.com/cockroachdb/cockroach/blob/5f75a1e6cf6f9a9e87f29f6a06b990a746bcf2d4/pkg/sql/logic_test.go#L1416)):*
> ```Go
> 	// We want to collect SQL perstatement statistics in tests,
> 	// regardless of what the environment / config says.
> 	defer settings.TestingSetBool(&sql.StmtStatsEnable, true)
> ```

add `()` at the end here too like tamir/peter says

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15361#-:-KifO3u82Fhq3gekes4T:bcxpqio)*

**Comment by bdarnell** [comment]
There's a typo in the commit message.

---

Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15361#-:-KifP91gBRLa3ynValRy:bm17vvy)*

**Comment by dt** [comment]
Fixed.

---

Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[pkg/kv/txn_coord_sender_test.go, line 1562 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/15361#-KifKjKD4Sn7HhmXg3ay-r1-1562:-KifPpQU4BZnlntSUydj:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5f75a1e6cf6f9a9e87f29f6a06b990a746bcf2d4/pkg/kv/txn_coord_sender_test.go#L1562)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

I think you're missing a pair of parens on the end of this.
</blockquote></details>

Done.

---

*[pkg/settings/registry.go, line 290 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/15361#-KifNglrdLXCOWT6RQk3:-KifPVhPiUbuaQEutPZx:bqjr0pa) ([raw file](https://github.com/cockroachdb/cockroach/blob/4129fe0eded7d8ad5bd34b20eb8c00331aaa0cf3/pkg/settings/registry.go#L290)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

What was the `setToDefault` doing? Or why is it no longer necessary?
</blockquote></details>

it was copying the `defaultValue` to `v` (via atomic) but we're just initializing `v` directly now (we don't need atomic yet, as we're still holding the only reference for now).

---

*[pkg/sql/logic_test.go, line 1416 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/15361#-KifNpI229iSWRrgk8xH:-KifPUXOpMyzjDrB0OXj:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5f75a1e6cf6f9a9e87f29f6a06b990a746bcf2d4/pkg/sql/logic_test.go#L1416)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

add `()` at the end here too like tamir/peter says
</blockquote></details>

Done.

---

*[pkg/storage/replica_test.go, line 7713 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/15361#-KifNZft83gmXPR4qOKn:-KifPrvIVEYvFwOebrIU:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5f75a1e6cf6f9a9e87f29f6a06b990a746bcf2d4/pkg/storage/replica_test.go#L7713)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

You need to invoke the function returned by `TestSetByteSize`.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15361#-:-KifPpnnuitFd6fRf1Y9:bt2vr1b)*

**Review by knz — APPROVED** [review]

**Pull request merged**